test-packages/memory-tests/sdk-canary/test-script.js

```diff
@@ -0,0 +1,115 @@
+const memwatch = require('node-memwatch-new');
+const { writeFileSync } = require('fs');
+const { resolve } = require('path');
+const { testService } = require('./test-service');
+
+const dest = { url: 'http://localhost:4004/' };
+
+const test = async () => {
+  // The script is written in js (not ts) and executed with --inspect flag because of this lib.
+  const hd = new memwatch.HeapDiff();
+  // It is intended to call execute multiple times without using a loop, so the deserializer, the entity builder and the schema will still be kept in the memory.
+  const testEntityApi = testService().testEntityApi;
+  const testEntities101 = await testEntityApi
```

Comment:
As Marika pointed out I would use a loop and keep the testEntities in an array, so that the garbage collector can not clean them away

Reply:
Also if you use a larger array perhaps you can use a already existing service?

Reply:
> Also if you use a larger array perhaps you can use a already existing service?

I would prefer using a big entity, so you can clearly understand after one request, what's the delta changes in the memory.
If you only use the existing entity, everything seems to be flat.